bin/summarizePerformanceRst.py

```diff
@@ -123,68 +127,130 @@ def plotMetricScatter(df, name1, name2, band,
+
+    data = astropy.table.vstack(list(per_tract.values()))
+
+    filters = set(data['Filter'])
+
+    df = data.to_pandas()
+    df.set_index(pd.MultiIndex.from_arrays([df.Metric, df.Filter]), inplace=True)
+
+    descriptions = {
+        "AM1": "repeatability (marcsec) for pairs at 5 arcmin",
+        "AM2": "repeatability (marcsec) for pairs at 20 arcmin",
+        "AF1": "outlier fraction (%) for pairs at 5 min",
+        "AF2": "outlier fraction (%) for pairs at 20 min",
+        "PA1": "repeatability (mmag) of PSF source magnitudes",
+        "PF1": "outlier fraction (%) deviating by more than PA2"
+    }
+
+    if args.plot:
+        for filt in filters:
+            plotMetricScatter(data, df, "AM1", "AF1", filt, descriptions, xmin=0, ymin=0)
+            plotMetricScatter(data, df, "AM2", "AF2", filt, descriptions, xmin=0, ymin=0)
+            plotMetricScatter(data, df, "PA1", "PF1", filt, descriptions)
+            plotMetricScatter(data, df, "AM1", "AF1", filt, descriptions, fromSingle=True, xmin=0, ymin=0)
+            plotMetricScatter(data, df, "AM2", "AF2", filt, descriptions, fromSingle=True, xmin=0, ymin=0)
+            plotMetricScatter(data, df, "PA1", "PF1", filt, descriptions, fromSingle=True)
+            plotMetricScatter(data, df, "AM1", "AF1", filt, descriptions,
+                              fromSingle=True, toMosaic=True, xmin=0, ymin=0)
+            plotMetricScatter(data, df, "AM2", "AF2", filt, descriptions,
+                              fromSingle=True, toMosaic=True, xmin=0, ymin=0)
+            plotMetricScatter(data, df, "PA1", "PF1", filt, descriptions, fromSingle=True, toMosaic=True)
+
     print()
+    print("jointcal calibrations that exceed metrics for a given filter+tract")
+    print("------------------------------------------------------------------")
+
+    # not including "PA1" metric here, since it's always above the spec
+    for metric in ("AM1", "AF1", "AM2", "AF2"):
+        print("Metric:", metric)
+        print("-----------")
+        test = data['Metric'] == metric
+        limit = data[test]['Design']
+        exceed = data[test]['Value_jointcal'] >= limit
+        for x in data[test][exceed]:
+            print("{} {} : {} > {}".format(x['Filter'], x['tract'], x['Value_jointcal'], x['Design']))
+        print()
+
+    def rms(x):
+        """Compute the root mean squared of a distribution."""
+        return np.sqrt(np.mean(x**2))
+
+    def print_y_is_less(x, y, name, verbose=False):
+        """Print a green `>` if y is less than x, otherwise a red `<`."""
+        if (verbose or x < y):
+            print(name, x, "\033[92m>\033[0m" if x > y else "\033[91m<\033[0m", y)
+
+    def print_y_is_less2(x, N, sigma, y, name, verbose=False):
+        """Print a green `>` if y is less than x+N*sigma, otherwise a red `<`."""
+        threshold = x + N*sigma
+        if (verbose or threshold < y):
+            less = "\033[92m>\033[0m" if threshold > y else "\033[91m<\033[0m"
```

Comment:
Color codes seem like a good place to use f-strings for clarity.

Reply:
Hmm... I think I came up with a neat solution there. Hadn't really used f-strings before, but I can see the utility.